### ISSUE TITLE
Ambient Occlusion broken in new patch.

### ISSUE THREAD

**Issue by frogles** [Lyall/MetaphorFix#98]
seems like all objects cast blob-like shadows around them now.. 

AO on:
<img width="1280" alt="ex1" src="https://github.com/user-attachments/assets/7196e700-ac52-4de2-9030-8f0406d01a87">

AO off:
<img width="1280" alt="ex2" src="https://github.com/user-attachments/assets/baf9e2a8-a416-4dbb-a1e4-32e14d87111f">

changing rendering scale minimizes this effect but the blobs come back after changing screens.

**Comment by Sharkcruiser** [comment]
Can confirm, even if its verly slightly visible with the right settings it is definitely there and can be quite distracting once you know its there :/ 
Definitely should be looked at

**Comment by system2ckc** [comment]
Was this not always the case?

**Comment by frogles** [comment]
> Was this not always the case?

it wasn't. not the greatest examples but these are the only two i have before the patch:

![2679460_20241011040814_1](https://github.com/user-attachments/assets/8dff423b-7d9b-4631-b76b-8df81da37798)

![2679460_20241012160217_1](https://github.com/user-attachments/assets/2f2458ed-3809-4e67-b747-cbc02e31b282)

**Comment by system2ckc** [comment]
> > Was this not always the case?
> 
> it wasn't. not the greatest examples but these are the only two i have before the patch:
> 

Hmm, for me I've always noticed the fuzzy AO underneath characters. So far it looks about the same, i'm bout to head to another hub so I'll test more once I get there. 

This is a screenshot from previous patch. See the fuzzy AO underneath the characters feet?
It does look alot more of a circular blob on your OP but maybe it's area dependent?
But yeah, I'm bout to be a new hub soon so I'll take a look again once I get there.

![image](https://github.com/user-attachments/assets/1f96bf5b-6955-42db-ad60-4b48a4109a86)

**Comment by frogles** [comment]
> Hmm, for me I've always noticed the fuzzy AO underneath characters. So far it looks about the same, i'm bout to head to another hub so I'll test more once I get there.
> 
> This is a screenshot from previous patch. See the fuzzy AO underneath the characters feet? It does look alot more of a circular blob on your OP but maybe it's area dependent? But yeah, I'm bout to be a new hub soon so I'll take a look again once I get there.

yeah i think you're right. i restarted the game and the beginning desert area where i took that previous screenshot doesn't have the issue at all whereas it's pretty minor in grand trad.

**Comment by Randy28CR** [comment]
Tbh, ambient occlusion in this game is not worth enough. It cost a lot GPU wise and, in my humble opinion, we should look for higher res to deal with the AA (or lack of it).
However, it could be cool if this can be address for preservation

**Comment by ipisano** [comment]
> Tbh, ambient occlusion in this game is not worth enough. It cost a lot GPU wise and, in my humble opinion, we should look for higher res to deal with the AA (or lack of it). However, it could be cool if this can be address for preservation

You made me disable ambient occlusion, and frankly speaking when it's applied to objects that already have shadows it only ruins the graphical presentation, for two reasons. 
First, in most cases it's inaccurate and grainy, even at higher resolution (I'm playing on a 4K display at 5K resolution with DLDSR) Secondly, as soon as you start moving the camera or god forbid the character, the AO starts "moving" and shimmering, in a game that already suffers from aliasing and shimmering... Considering this, screenshots won't be a good comparison medium; test on your own, and be sure to move around.

Then again, I also noticed some objects that completely lack shadows, and AO helps with those. But is the performance hit and the blob under the character worth it? I know we can reduce the AO's resolution using the fix, but the AO quality for me was already severely lacking at 5K, setting the scale at 0.75 to get it back to my display's native 4K does help with performance but makes the AO look even worse.

In the end I decided to disable it, I think the resulting image is much more stable and clean, when I'm actually playing the game I'm sure to notice the added graininess, shimmering that AO even without trying to look with a critical eye, whereas I probably won't notice a couple of objects not having shadows.

**Comment by Randy28CR** [comment]
> > Tbh, ambient occlusion in this game is not worth enough. It cost a lot GPU wise and, in my humble opinion, we should look for higher res to deal with the AA (or lack of it). However, it could be cool if this can be address for preservation
> 
> You made me disable ambient occlusion, and frankly speaking when it's applied to objects that already have shadows it only ruins the graphical presentation, for two reasons. First, in most cases it's inaccurate and grainy, even at higher resolution (I'm playing on a 4K display at 5K resolution with DLDSR) Secondly, as soon as you start moving the camera or god forbid the character, the AO starts "moving" and shimmering, in a game that already suffers from aliasing and shimmering... Considering this, screenshots won't be a good comparison medium; test on your own, and be sure to move around.
> 
> Then again, I also noticed some objects that completely lack shadows, and AO helps with those. But is the performance hit and the blob under the character worth it? I know we can reduce the AO's resolution using the fix, but the AO quality for me was already severely lacking at 5K, setting the scale at 0.75 to get it back to my display's native 4K does help with performance but makes the AO look even worse.
> 
> In the end I decided to disable it, I think the resulting image is much more stable and clean, when I'm actually playing the game I'm sure to notice the added graininess, shimmering that AO even without trying to look with a critical eye, whereas I probably won't notice a couple of objects not having shadows.

My finding were not as deep as yours, however, I just felt like the game had something wrong with AO in movement. Only noticeable place where you can actually tell it working as intended is at nights, when under a public lamp. Beyond that, I felt with game shadows was more than enough.
You can turn AO off and bump resolution, which actually helps to get rid of shimmery.
Currently playing at 4k on an RTX 3070 with res scale up to x1.25. Silky smooth and not a very perceptible shimmery. 
With AO that wouldn't be posible and for nothing, pretty much

**Comment by Lyall** [comment]
Is this caused by MetaphorFix when adjusting the AO resolution? Or is it a newly introduced game bug in the latest update?

**Comment by system2ckc** [comment]
Im pretty sure its just the game and its always been this way. But the AO will look a bit different depending on the area.

**Comment by Lyall** [comment]
Yeah I was going to say that the AO tends to look pretty poor even at 200% resolution scale, so I'm not surprised there are issues in the vanilla game in later areas. If it's not something that MetaphorFix is causing, I'm not sure what could be done to improve it. It probably warrants a bit more investigation. I do however wonder what is going wrong in this example where AO behaviour completely changes.
